### PR TITLE
Refactor/remove product stocks

### PR DESCRIPTION
**Pull request by eschrewe**
- removed classes related to the previous product-stock-api (including: entities, services, repos, dtos), which are no longer in use now, solving #175 
- removed puris.apiversion env variable
- moved ItemUnitEnumeration to the stock/model/measurement package

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/api/logic/service/VariablesService.java [hunk 1]
```diff
@@ -46,11 +46,6 @@ public class VariablesService {
      * defined in the DataInjectionCommandLineRunner
      */
     private String demoRole;
-    @Value("${puris.apiversion}")
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/common/api/logic/service/VariablesService.java:49]: Does it overall make sense to move the VariableService out of the common.api package into something like common.util?

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/api/domain/model/datatype/DT_RequestStateEnum.java [hunk 2]
```diff
@@ -20,13 +20,9 @@
  */
 package org.eclipse.tractusx.puris.backend.common.api.domain.model.datatype;
 
-import org.eclipse.tractusx.puris.backend.stock.domain.model.ProductStockRequest;
-import org.eclipse.tractusx.puris.backend.stock.domain.model.ProductStockResponse;
-
 /**
  * Enum to track the status of
- * {@link ProductStockRequest} and
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/common/api/domain/model/datatype/DT_RequestStateEnum.java:28]: Please update license header

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/domain/model/measurement/ItemUnitEnumeration.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-package org.eclipse.tractusx.puris.backend.stock.logic.dto.itemstocksamm;
+package org.eclipse.tractusx.puris.backend.stock.domain.model.measurement;
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/domain/model/measurement/ItemUnitEnumeration.java:20]: Please update license header

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapperTest.java [hunk 4]
```diff
@@ -29,6 +29,7 @@
 import org.eclipse.tractusx.puris.backend.stock.domain.model.ItemStock;
 import org.eclipse.tractusx.puris.backend.stock.domain.model.MaterialItemStock;
 import org.eclipse.tractusx.puris.backend.stock.domain.model.ReportedProductItemStock;
+import org.eclipse.tractusx.puris.backend.stock.domain.model.measurement.ItemUnitEnumeration;
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapperTest.java:32]: Please update license header

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Overall great work, please check question and update license headers

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/api/logic/service/VariablesService.java (hunk 1 above)
Comment (eschrewe) [reply backend/src/main/java/org/eclipse/tractusx/puris/backend/common/api/logic/service/VariablesService.java:49]: Yes, I agree. I'd suggest the same for the PatternStore

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/FrontendMaterialDto.java [hunk 5]
```diff
@@ -26,7 +26,7 @@
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
-import org.eclipse.tractusx.puris.backend.common.api.logic.service.PatternStore;
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/FrontendMaterialDto.java:29]: Please update license header

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/ProductStockRequestForMaterialDto.java [hunk 6]
```diff
@@ -25,7 +25,7 @@
 import jakarta.validation.constraints.NotNull;
 import jakarta.validation.constraints.Pattern;
 import lombok.*;
-import org.eclipse.tractusx.puris.backend.common.api.logic.service.PatternStore;
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/ProductStockRequestForMaterialDto.java:28]: Please update license header

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
After moving VariablesService & PatternStore, two more files need license header update. Please update.

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/ProductStockRequestForMaterialDto.java (hunk 6 above)
Comment (tom-rm-meyer-ISST) [reply backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/ProductStockRequestForMaterialDto.java:28]: true

**Review by tom-rm-meyer-ISST — APPROVED** [review]
LGTM, Change contains only deletions which are not critical to IP issues from my point of view as the functionality has been superseeded by new standardization candidate

**Pull request merged**